### DISCUSSION TITLE
Strapi good for Production web applications?

### DISCUSSION THREAD

**Discussion by piyushpaliwal** [strapi/strapi#6847] [Legacy Topics]
Hi, I am on the lookout for a platform or framework which can help me manage entire backend for a web application which will be used by ~20k people per day.
Based on what I am reading, it has the potential to manage DB, build complex data types and provide a Admin interface to interact with data **BUT** all of this is with regards to websites and not web applications.

Is strapi good enough for the use case?

**Comment by iicdii** [comment]
I am running a small web application concurrently used by 50 people. There were some limitations while developing, but most of the functions can be customized by the developer. I'd recommend you to read #6634, users are explaining the limitations of Strapi well.

I think the biggest problem is the ORM. **Strapi's not completely optimized for queries yet.** If you designed a complex model, you will have to make a lot of effort to reduce the cost of the query(Unnecessary join operation causes bad performance). I use most queries as mongoose custom queries instead of using Strapi ORM.

If you don't have enough back-end development knowledge or **if you want to reduce the configuration time of the development environment, Strapi will be a good choice for you.** I was able to implement most requirements in about a month using the Strapi + Next.js. It's a great advantage is that you can quickly implement time-consuming jobs such as authentication, authority, email, and file uploads with simple settings.
